### PR TITLE
Move clock constraints from JSON to PCF

### PR DESCRIPTION
**Pull request by tjurtsch**
This PR solves issue https://github.com/QuickLogic-Corp/symbiflow-arch-defs/issues/238

It adds a new PCF directive "set_clk <pin> <net>" and updates repacker class with PCF-reading function.
Also the "counter_4clk" test was updated to match the new constrain scheme.

### CODE REVIEW

File: quicklogic/qlf_k4n8/utils/repacker/repack.py [hunk 1]
```diff
@@ -929,6 +930,48 @@ def load_repacking_constraints(json_root):
...
+            constraints.append(constraint)
+
+            logging.debug(
+                "  {}: {}.{}[{}]".format(
+                    constraint.net, constraint.block_type, constraint.port,
+                    constraint.pin
+                )
+            )
+
+            constraint.block_type = 'io'
```
Comment (mkurc-ant) [inline quicklogic/qlf_k4n8/utils/repacker/repack.py:959]: I think you have to create a new `RepackingConstraint` object here. Right now this code changes the block type of the existing constraints and adds it again to the list so in the end there are two identical.

**Review by mkurc-ant — CHANGES_REQUESTED** [review]

File: quicklogic/qlf_k4n8/utils/repacker/repack.py (hunk 1 above)
Comment (tjurtsch) [reply quicklogic/qlf_k4n8/utils/repacker/repack.py]: Fixed

File: utils/lib/parse_pcf.py [hunk 2]
```diff
@@ -2,6 +2,7 @@
 import re
```
Comment (mkurc-ant) [inline utils/lib/parse_pcf.py:2]: Please add a docstring here describing the supported pcf commands (set_io and set_clk)

**Review by mkurc-ant — APPROVED** [review]
Please add the comment. Otherwise LGTM

**Comment by mkurc-ant** [comment]
@lnsharma @tpagarani Could you please review this PR?

**Review by lnsharma — APPROVED** [review]
Looks good. Please see if we need to update wrapper scripts considering that now we do not require json file for clock constraints.

File: utils/lib/parse_pcf.py (hunk 2 above)
Comment (tjurtsch) [reply utils/lib/parse_pcf.py:9]: Added

**Comment by kkumar23** [comment]
The CI is red . Can we fix and then merge this PR.

**Comment by kkumar23** [comment]
@mkurc-ant @lnsharma : should be have global clocks usage in the report file? As of now report file does not show any global clk used vs available.

**Pull request merged**